Ginger/GingerCore/Environments/Database.cs

```diff
@@ -494,7 +494,9 @@ public Boolean Connect(bool displayErrorPopup = false)
             }
             catch (Exception e)
             {
-                Reporter.ToLog(eLogLevel.ERROR, "DB connection failed, DB type: " + DBType.ToString() + "; Connection String =" + ConnectionString, e);
+                //Hide paasword from the ConnectionString
+                connectConnectionString = connectConnectionString.Replace(connectConnectionString.Substring(connectConnectionString.ToLower().IndexOf("password") + "password".Length), "=*****");
```

Comment:
1-This code change assum "password" will be last in the connection string, but is it always? can't password come before userName or other?

2- check how password appera for all common connection strings, for example for MySQL I see the password is "Pwd"
sample: "Server=myServerAddress;Database=myDataBase;Uid=myUsername;Pwd=myPassword;"

one approach can be to remove all spaces from connection string and then look for for "password=" or "pwd=" (or other password string types if there is) end index and then to replace all between this index to ";" or end of string with "****"

Reply:
Done. Please review